task/trivy-scanner/0.2/trivy-scanner.yaml

```diff
@@ -0,0 +1,60 @@
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: trivy-scanner
+  labels:
+    app.kubernetes.io/version: "0.1"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/categories: Security
+    tekton.dev/tags: CLI, trivy
+    tekton.dev/displayName: "trivy scanner"
+    tekton.dev/platforms: "linux/amd64"
+spec:
+  description: >-
+    Trivy is a simple and comprehensive scanner for
+    vulnerabilities in container images,file systems
+    ,and Git repositories, as well as for configuration issues.
+
+    This task can be used to scan for vulnenrabilities on the source code
+    in stand alone mode.
+  workspaces:
+    - name: manifest-dir
+  params:
+    - name: ARGS
+      description: The Arguments to be passed to Trivy command.
+      type: array
+    - name: TRIVY_IMAGE
+      default: docker.io/aquasec/trivy@sha256:dea76d4b50c75125cada676a87ac23de2b7ba4374752c6f908253c3b839201d9
+      description: Trivy scanner image to be used
+    - name: IMAGE_PATH
+      description: Image or Path to be scanned by trivy.
+      type: string
+    - name: AIR_GAPPED_ENABLED
+      default: "false"
+      description: a flag enabling Air-Gapped mode
+      type: string
+  steps:
+    - name: trivy-scan
+      image: $(params.TRIVY_IMAGE)
+      workingDir: $(workspaces.manifest-dir.path)
+      script: |
+        #!/usr/bin/env sh
+          cmd="trivy $* "
+          if [ "$(params.AIR_GAPPED_ENABLED)" = "true" ]; then
+            echo "Air-Gapped mode enabled"
+            TRIVY_TEMP_DIR=$(mktemp -d)
+            trivy --cache-dir $TRIVY_TEMP_DIR image --download-db-only
+            tar -cf ./db.tar.gz -C $TRIVY_TEMP_DIR/db metadata.json trivy.db
+            rm -rf $TRIVY_TEMP_DIR
```

Comment:
I am suspecting that an organisation will have to enable traffic atleast for once in order to download the db, right?

Reply:
Yep, this is not a totally network-free solution, I know you assumed that way, as we said, `air-gapped.`

Reply:
Maybe is there a way to provide a PVC with already downloaded DB or to use a PROXY to download that DB?

Reply:
@keskad, not sure about the proxy part but for the other part, ie, way to provide a PVC, yes, there can be 2 ways:
1. If size of downloaded db is less than 1MB then one can use ConfigMap
OR
2. Create a PVC, first mount it to a Pod and use `kubectl cp` command to copy the downloaded db from your local machine to pvc and then subsequently use that pvc wherever you want https://www.middlewareinventory.com/blog/kubectl-cp-example/